python-sdk/noxfile.py

```diff
@@ -42,6 +42,9 @@ def test(session: nox.Session, airflow) -> None:
         )
         session.install(f"apache-airflow=={airflow}", "-c", constraints_url)
         session.install("-e", ".[all,tests]", "-c", constraints_url)
+        # install apache-airflow-providers-sftp==4.0.0 since it is the compatible version
+        # to run sftp example dag to load file with airflow 2.2.5
+        session.install("apache-airflow-providers-sftp==4.0.0")
```

Comment:
```suggestion
        session.install("apache-airflow-providers-sftp>=4.0.0")
```

Reply:
No we can't add  make greater than since it will install 4.2.0 and that supports airflow 2.3 and above. 
Here we need sftp which is compatible with 2.2.5 which is 4.0.0.

Reply:
We should document this somewhere in our docs

Reply:
hmm 🤔 The pip resolver should take care of this together with the constraints. How can it install a version which is not allowed for the currently install airflow version?

We should not need the `session.install` here at all.

Reply:
by default in constraints file its `apache-airflow-providers-sftp==2.5.0` so it doesn't allow anything other than 2.5.0. But i would need 4.0.0 which works with 2.2+.